pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -258,10 +277,55 @@ func (b *ServiceBinder) Bind() (reconcile.Result, error) {
 	}
 	sbrStatus.Secret = secretObj.GetName()
 
+	if isApplicationSelectorEmpty(b.SBR.Spec.ApplicationSelector) {
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:   CollectionReady,
+			Status: corev1.ConditionTrue,
+		})
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:    InjectionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  EmptyApplicationSelectorReason,
+			Message: ErrEmptyApplicationSelector.Error(),
+		})
+
+		// updating status of request instance
+		sbr, err := b.updateStatusServiceBindingRequest(b.SBR, sbrStatus)
+		if err != nil {
+			return RequeueOnConflict(err)
+		}
+
+		// appending finalizer, should be later removed upon resource deletion
+		sbr.SetFinalizers(append(removeStringSlice(b.SBR.GetFinalizers(), Finalizer), Finalizer))
+		if _, err = b.updateServiceBindingRequest(sbr); err != nil {
+			return NoRequeue(err)
+		}
+
+		b.Logger.Info(ErrEmptyApplicationSelector.Error())
+		return Done()
```

Comment:
This block deserves its own function:

```go
if isApplicationSelectorEmpty(b.SBR.Spec.ApplicationSelector) {
        return b.handleApplicationSelectorEmpty(b.SBR)
}
```

This modification has two important benefits:

* Introduces more coverage by making it easier to cover the intended use cases; and
* Provide supporting material in further refactoring and simplifications.

What do you think?

Reply:
Agreed.

Application Not Found and Empty Application scenario should be handled in different function.

Done 1b275f9